src/iplugingame.h

```diff
@@ -170,6 +170,13 @@ class IPluginGame : public QObject, public IPlugin
    */
   virtual QDir dataDirectory() const = 0;
 
+  /**
+   * this function may be called before init()
+   *
+   * @return directories where we may find data files outside the main location
+   */
+  virtual QMap<QString, QDir> secondaryDataDirectories() const = 0;
```

Comment:
This doesn't really feel like it's something that we can reuse for many games. Probably only for starfield. As such I would suggest we put this into a GameFeature so we don't pollute the main IPluginGame interface.

I think it might've been cleaner to add this way too.

Reply:
I don't really mind the extra functions for `IGamePlugin`, moving it to a game feature seems overkill for a single function. But I think this should be defaulted instead of pure-virtual.

Reply:
Ok, then we can keep it like this. I would consider moving this to a dedicated feature if we find we need to expand this with more methods in the future.

Reply:
I also think this should return an `std::map` instead of a `QMap`. Most recent changes to `uibase` use standard containers instead of Qt ones.

Reply:
I could replace the original function, but then we need to adjust every game plugin. This still provides functionality that could be potentially useful to other games without requiring updates to every other game plugin. Though I suppose you could deprecate it and have gamebryo handle converting the old method to a new map.

In any case, it can be updated in the future.

Reply:
Sometimes games have their own VFS, so long-term, it could be nice if we had something both for dealing with that and with Starfield. It would probably be reasonable to move to a dedicated game feature if we implement support for mutable VFSes, i.e. where we can install things with them instead of using USVFS.